compiler/rustc_trait_selection/src/solve/eval_ctxt/canonical.rs

```diff
@@ -365,8 +365,8 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
 }
 
 /// Resolves ty, region, and const vars to their inferred values or their root vars.
-struct EagerResolver<'a, 'tcx> {
-    infcx: &'a InferCtxt<'tcx>,
+pub struct EagerResolver<'a, 'tcx> {
```

Comment:
please instead move the `EagerResolver` to somewhere else, either `rustc_infer` or idk if `rustc_ty_utils` or whatever works :grin: anyways, to any place which is not `solve` :sparkles: